Assets/Source/Components/UI/CanvasMenuSelectorComponent.cs

```diff
@@ -2,11 +2,23 @@
 using Assets.Source.Components.UI.Base;
 using System.Linq;
 using UnityEngine;
+using UnityEngine.EventSystems;
 
 namespace Assets.Source.Components.UI
 {
     public class CanvasMenuSelectorComponent : ComponentBase
     {
+
+        public override void ComponentAwake()
+        {
+            if (FindObjectOfType<EventSystem>() == null)
+            {
+                var eventSystem = new GameObject("EventSystem", typeof(EventSystem), typeof(StandaloneInputModule));
+                EventSystem.current = eventSystem.GetComponent<EventSystem>();
+            }
```

Comment:
nice fam, just in case i delete more event systems it'll still work haha. I actually like that we are generating most things in the level rather than relying on them being manually setup

Reply:
yuss i actually had this happen to me.  Turns out that canvas only gets created when a ui element is instantiated, not the canvas.  So sometimes it ended up not being created on time.  This makes sure it is